stetho/src/main/java/com/facebook/stetho/common/ListUtil.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.common;
+
+import java.util.AbstractList;
+import java.util.Collections;
+import java.util.List;
+
+public final class ListUtil {
+  private ListUtil() {
+  }
+
+  /**
+   * Compares two {@link List}s by comparing their contained items by object identity.
+   */
+  public static <T> boolean identityEquals(List<? extends T> list1, List<? extends T> list2) {
+    if (list1 == list2) {
+      return true;
+    }
+    int size = list1.size();
+    if (size != list2.size()) {
+      return false;
+    }
+    for (int i = 0; i < size; ++i) {
+      if (list1.get(i) != list2.get(i)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Copies the given {@link List} and returns the copy as an immutable {@link List}.
+   */
+  public static <T> List<T> copyToImmutableList(List<T> list) {
```

Comment:
... and after thinking on this a bit more, I still think this is a reasonable approach. It's small and to the point, and enables a lot of simplicity over in ElementInfo and ShadowDOM while also saving 1 allocation on almost every ElementInfo instance. The big problem I've seen with ImmutableList elsewhere (at least on Android) is how it's so casually used and ends up encourage lots of unnecessary copying (e.g. allocating an array, passing it to of(T...) which then copies the array even though just wrapping the original array would've worked fine because it isn't used thereafter). I'm not using this casually; I'm being very intentional.